src/libAtomVM/term.h

```diff
@@ -1079,4 +1079,28 @@ static inline int term_is_number(term t)
  */
 void term_display(FILE *fd, term t, const Context *ctx);
 
+/**
+ * @brief Checks if a term is a string (i.e., a list of characters)
+ *
+ * @details Returns 1 if a term is a list of characters or an empty list (nil term), otherwise 0.
+ * @param t the term that will be checked.
+ * @return 1 if check succedes, 0 otherwise.
+ */
+static inline int term_is_string(term t)
+{
+    if (term_is_list(t)) {
+        term i = t;
+        while (!term_is_nil(i)) {
+            term e = term_get_list_head(i);
+            if (!term_is_uint8(e)) {
+                return 0;
+            }
+            i = term_get_list_tail(i);
+        }
```

Comment:
I added a check up front to check if the list is proper.  If not, the function return 0 immediately.  Otherwise, it enters this loop (using `term_is_nonemtpy_list`).  That should be safe, right?

Reply:
yes, right